include/fcc_bridge_node.hpp

```diff
@@ -125,155 +195,766 @@ class FCCBridgeNode : public common_lib::CommonNode {
+    /*************************************************************************/
 
-    // Cached ROS messages
     interfaces::msg::Heartbeat
         last_mission_control_heartbeat; /**< The last received heartbeat from
                                           mission control */
 
+    /**************************************************************************/
+    /*                             Safety members                             */
+    /**************************************************************************/
+
+   private:
+    // Safety limits
+    struct safety_limits {
+        // TODO: Implement
+        constexpr static float MIN_SPEED_LIMIT_MPS =
+            0; /**< The minimum speed allow when flying to a waypoint. Exclusive
+                */
+        constexpr static float HARD_MAX_SPEED_LIMIT_MPS =
+            5; /**< The hard speed limit which will cap the soft speed limit.
+                  Inclusive */
+        float max_speed_mps;
+    }; /**< struct to hold all currently active safety limits */
+
+    std::optional<struct safety_limits>
+        safety_limits; /**< Safety limits to be enforced such as geofence and
+                          max speed */
+
+    /**************************************************************************/
+    /*                            Safety functions                            */
+    /**************************************************************************/
+
+   private:
+    /**
+     * @brief Callback function to be triggered when an asynchronous RTH mavsdk
+     * action finished
+     *
+     * @param result The result of the RTH
+     *
+     * @note Will exit the process if the result code does not indicate success
+     *
+     * Sets FCCBridgeNode::internal_state to
+     * FCCBridgeNode::INTERNAL_STATE::LANDED if result indicated success
+     *
+     * Implemented in src/fcc_bridge_node_safety.cpp
+     */
+    void mavsdk_rth_cb(const mavsdk::Action::Result &result);
+
    protected:
-    // Safety functions
-    // bool check_point_in_geofence();
+    /**
+     * @brief Checks if the current GPS Fix type is adequate for the current
+     * internal state
+     *
+     * Verifies the GPS fix if the drone is armed and has not yet landed. If no
+     * 2DFix is available abort and exit.
+     * TODO: Is a 2D fix necessary or can we always assume a 3D fix
+     *
+     * Exits if no GPS is installed.
+     *
+     * TODO: Geofence check
+     *
+     * @throws std::runtime_error If FCCBridgeNode::internal_state is
+     * FCCBridgeNode::INTERNAL_STATE::STARTING_UP,
+     * FCCBridgeNode::INTERNAL_STATE::ROS_SET_UP, or
+     * FCCBridgeNode::INTERNAL_STATE::ERROR
+     *
+     * Implemented in src/fcc_bridge_node_safety.cpp
+     */
+    void check_gps_state();
+    /**
+     * @brief Checks if the current flight state is adequate for the current
+     * internal state
+     *
+     * TODO: Currently just an empty function
```

Comment:
My goal is to figure out what a function does on a high level and document it inside the docstring.
This way I only have to look at the docstring for the rough functionality instead of the code.
I do this even when the function is not yet fully implemented.